tests/transforms/test_qcut.py

```diff
@@ -2917,3 +2920,152 @@ def test_validate_fragment_sizes(self, max_wires_by_fragment, max_gates_by_fragm
                 max_wires_by_fragment=max_wires_by_fragment,
                 max_gates_by_fragment=max_gates_by_fragment,
             )
+
+
+def make_weakly_connected_tape(
+    fragment_wire_sizes=[3, 5],
+    single_gates_per_wire=1,
+    double_gates_multiplier=1.5,
+    inter_fragment_gate_wires={(0, 1): 1},
+    repeats=2,
+    seed=None,
+):
+    """Helper function for making random weakly connected tapes."""
+    rs = np0.random.RandomState(seed=seed)
+    inter_fragment_gate_wires = inter_fragment_gate_wires or {}
+    with qml.tape.QuantumTape() as tape:
+        for _ in range(repeats):
+            for i, wire_size in enumerate(fragment_wire_sizes):
+                double_gates_per_fragment = int(double_gates_multiplier * wire_size)
+                for _ in range(double_gates_per_fragment):
+                    j0, j1 = rs.choice(wire_size, size=2, replace=False)
+                    qml.CNOT(wires=[f"{i}-{j0}", f"{i}-{j1}"])
+                for _ in range(single_gates_per_wire):
+                    for j in range(wire_size):
+                        qml.RZ(0.5, wires=f"{i}-{j}")
+            for frag_pair, num_gates in inter_fragment_gate_wires.items():
+                f0, f1 = sorted(frag_pair)
+                for _ in range(num_gates):
+                    w0 = rs.choice(fragment_wire_sizes[f0])
+                    w1 = rs.choice(fragment_wire_sizes[f1])
+                    qml.CNOT(wires=[f"{f0}-{w0}", f"{f1}-{w1}"])
+        for i, wire_size in enumerate(fragment_wire_sizes):
+            qml.expval(qml.PauliZ(wires=[f"{i}-0"]))
+    return tape
+
+
+class TestKaHyPar:
+    """Tests for the KaHyPar cutting function and utilities."""
+
+    # Fixes seed for Github actions:
+    seed = 11 if environ.get("CI") == "true" else None
```

Comment:
I would say I want to keep this line unless you force me 😁. It's an easy change with the `RandomState` is isolated, and I personally would like to experiment if this works as a middle ground for randomized tests.

Reply:
After looking at the seed fixing example for device test suite, I think I have a conjecture why earlier there's an unrelated test failing. If I remember correctly, `np.random.seed()` fixes its default internal random state, but each subsequent `np.random.*` calls changes the state. So if the order of these calls changes, or if new such calls got inserted/removed, the subsequent outputs will also change.

I'm not sure how much is shared in a pytest session, but since the failing test got fixed after I switched to vanilla numpy's `RandomState` which is an isolated instance, it could be that the previous `np.random.choice` calls disturbed the usual calling order of random functions that depended on the global random state, thus causing the failure.

Reply:
Assuming all developers know about this, or fix their randomized tests' seeds this way, then if one wants to run tests with fixed seed, one can simply simulate the CI environment: `CI=true pytest`.

This is likely not the best place for this discussion. But a slightly better solution is to use an environment variable to set a seed from which all randomized tests load. So that calling `pytest` will do truly randomized tests and `PL_TEST_SEED=99 pytest` will run the tests with global seed 99.